### DISCUSSION TITLE
GeoParquet 1.1.0 covering bounding box precision

### DISCUSSION THREAD

**Discussion by mdjong1** [geopandas/geopandas#3396] [General]
Hi there 👋🏼 

We've recently been writing a lot of datasets using the `write_covering_bbox` parameter when writing GeoParquet files using the 1.1.0 spec. First of all, amazing work on supporting that, we're super happy with it and hope to be releasing an open source dataset using the spec which would not have been possible without geopandas.

My main question is in relation to the precision of the covering bounding box; I've noticed that it's created with a lot of decimal places, and I was wondering whether it would make sense to either:
1. Have control over the bounding box precision as an argument to `to_parquet()`
2. or; Limit the precision by default

Example of a bounding box where I'm getting a few more decimals than I think are necessary
<img width="550" alt="afbeelding" src="https://github.com/user-attachments/assets/d613794f-57ed-4c13-9c9a-e0f7ecee649c">

To reference the great XKCD comic on coordinate precision:
![afbeelding](https://github.com/user-attachments/assets/628a5ee5-8b46-4e1c-b06a-88e2144ec16d)
https://xkcd.com/2170/

**Comment by m-richards** [comment]
@mdjong1 I think from the geopandas perspective, the most important thing is that the covering bounding box actually matches the extent of your data. Are you seeing a situation where you're seeing more precision in the bounding box than the data you are writing to the parquet?

```python
import json

from pyarrow import parquet
import geopandas as gpd
import pandas as pd

df = gpd.GeoDataFrame(geometry=gpd.GeoSeries.from_xy([0,10/7], [1,1]))
filename = "test.parquet"
df.to_parquet(filename, write_covering_bbox=True)
df_read = pd.read_parquet((filename))
print(pd.json_normalize(df_read['bbox']))
       xmin  ymin      xmax  ymax
0  0.000000   1.0  0.000000   1.0
1  1.428571   1.0  1.428571   1.0
```
Here the bbox outputs match the precision of the input values - to me it doesn't make sense to limit the precision of the covering bbox if you're not limiting the precision of the coordinates - because then the xkcd comic is just as relevant to the geometry itself.

If you want to treat both equivalently you can use `set_precision`:
```python
df = gpd.GeoDataFrame(geometry=gpd.GeoSeries.from_xy([0,10/7], [1,1]))
filename = "test.parquet"
df.geometry = df.geometry.set_precision(0.1)
df.to_parquet(filename, write_covering_bbox=True)

df_read = pd.read_parquet((filename))
print(pd.json_normalize(df_read['bbox']))
   xmin  ymin  xmax  ymax
0   0.0   1.0   0.0   1.0
1   1.4   1.0   1.4   1.0

```

**Reply by mdjong1** [reply]
Hi @m-richards, thanks for the reply! That actually makes a lot of sense, and I totally forgot to think of geopandas in that way. I was converting some third-party data and neglected to check the precision of the underlying data, which is, indeed, at the same precision as geopandas writes it to the covering bounding box.

I'll drop the precision of the underlying geometry prior to writing the outputs in the future; thank you!
